plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -270,6 +277,25 @@ data class ClassTypeEmbedding(val className: ScopedKotlinName) : TypeEmbedding {
         return Predicate(name, listOf(subjectEmbedding.toLocalVarDecl()), body)
     }
 
+    fun getterFunctions(): List<Function> {
+        val receiver = VariableEmbedding(GetterFunctionSubjectName, this)
+        val getPropertyFunctions = fields.values
+            .filter { it.accessPolicy != AccessPolicy.ALWAYS_INHALE_EXHALE }
+            .map { FieldAccessFunction(name, it.toViper(), FieldAccess(receiver, it).toViper()) }
+        val getSuperPropertyFunctions = classSuperTypes.flatMap {
+            it.allFields
+                .filter { field -> field.accessPolicy != AccessPolicy.ALWAYS_INHALE_EXHALE }
+                .map { field ->
+                    FieldAccessFunction(
+                        name,
+                        field.toViper(),
+                        Exp.FuncApp(GetterFunctionName(it.name, field.name), listOf(receiver.toViper()), field.type.viperType)
+                    )
+                }
+        }
```

Comment:
It's a bit confusing that `it` is a field in the first case and a supertype in the second case, perhaps it would be best to change the naming.